opta/core/aws.py

```diff
@@ -167,6 +167,29 @@ def force_delete_terraform_lock_id(self) -> None:
             Key={"LockID": {"S": f"{bucket}/{self.layer.name}"}},
         )
 
+    @classmethod
+    def get_detailed_config_map(cls) -> Dict[str, Dict[str, str]]:
+        prefix = "opta_config/"
+        s3 = boto3.client("s3")
+        opta_config_detailed_map = {}
+        for aws_bucket in cls.get_bucket_list():
+            detailed_configs = {}
+            response = s3.list_objects(Bucket=aws_bucket, Prefix=prefix, Delimiter="/")
+            if "Contents" in response:
+                for data in response["Contents"]:
+                    config_object = s3.get_object(Bucket=aws_bucket, Key=data["Key"])
+                    detailed_configs[data["Key"][len(prefix) :]] = json.loads(
+                        config_object["Body"].read()
+                    )["original_spec"]
+                opta_config_detailed_map[aws_bucket] = detailed_configs
+        return opta_config_detailed_map
+
+    @classmethod
+    def get_bucket_list(cls) -> List[str]:
+        s3 = boto3.client("s3")
+        aws_bucket_data = s3.list_buckets().get("Buckets", [])
```

Comment:
this method can take a parameter called `Prefix` we should use it.
check if there is something similar for gcp

Reply:
We can think about this, but maybe in an incremental stage. When we have a concrete thing on how we want the bucket names to look like. Once we have that fixed, we can add some sort of prefix. But till that time, let's use the logic to find the `opta_config` directory to figure out if the current bucket is an opta bucket or not.